store/sqlstore/channel_store.go

```diff
@@ -299,21 +320,112 @@ func (s SqlChannelStore) CreateIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_channelmembers_user_id", "ChannelMembers", "UserId")
 
 	s.CreateFullTextIndexIfNotExists("idx_channel_search_txt", "Channels", "Name, DisplayName, Purpose")
+
+	s.CreateIndexIfNotExists("idx_publicchannels_team_id", "PublicChannels", "TeamId")
+	s.CreateIndexIfNotExists("idx_publicchannels_name", "PublicChannels", "Name")
+	s.CreateIndexIfNotExists("idx_publicchannels_delete_at", "PublicChannels", "DeleteAt")
```

Comment:
Did you verify that an index on `DeleteAt` improves query plans or times?

Reply:
No, but thanks for the reminder: at this point, the indices here mirror the existing `Channels` table so as to increase scope. I've filed https://mattermost.atlassian.net/browse/MM-12763 as a general purpose ticket to go back and review all indices in question and make sure we're not forcing the database to do unnecessary work, here or elsewhere.